### PR TITLE
feat: support for a general validation message in query editor

### PR DESCRIPTION
**Pull request by suryabhaskarkarri**

### CODE REVIEW

**Comment by suryabhaskarkarri** [comment]
UI screenshots on how the message would look like

Screenshot 2023-02-09 at 11 30 50 AM
Screenshot 2023-02-09 at 11 38 13 AM

File: querybook/server/lib/query_analysis/validation/base_query_validator.py [hunk 1]
```diff
@@ -15,18 +20,21 @@ class QueryValidationSeverity(Enum):
 class QueryValidationResult(object):
     def __init__(
         self,
+        obj_type: QueryValidationResultObjectType,
```
Comment (czgu) [inline querybook/server/lib/query_analysis/validation/base_query_validator.py:23]: can you make this optional and default to QueryValidationResultObjectType.LINT? this makes sure other plugin still works as is

File: querybook/server/lib/query_analysis/validation/base_query_validator.py [hunk 2]
```diff
@@ -7,6 +7,11 @@
...
 
 
+class QueryValidationResultObjectType(Enum):
+    LINT = "lint"
+    GENERAL = "general"
+
+
 class QueryValidationSeverity(Enum):
     ERROR = "error"
     WARNING = "warning"
```
Comment (jczhong84) [inline querybook/server/lib/query_analysis/validation/base_query_validator.py:17]: do we need the `INFO` severity for the runtime estimation?

**Comment by czgu** [comment]
@suryabhaskarkarri @jczhong84 what is your thought of calling the types `LOCAL` and `GLOBAL` instead of `LINT` and `GENERAL`?

**Comment by suryabhaskarkarri** [comment]
@czgu I feel LINT seems to be more meaningful & understandable.

**Review by czgu — APPROVED** [review]

**Pull request merged**